agent/reconciler/host_reconciler.go

```diff
@@ -28,14 +29,26 @@ import (
 	infrastructurev1beta1 "github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/apis/infrastructure/v1beta1"
 )
 
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+//counterfeiter:generate . Installer
+type Installer interface {
+	Install(string, string, string, logr.Logger) error
+	Uninstall(string, string, string, logr.Logger) error
+}
+
+type K8sInstaller struct {
+	DownloadPath string
+}
+
 type HostReconciler struct {
 	Client           client.Client
 	CmdRunner        cloudinit.ICmdRunner
 	FileWriter       cloudinit.IFileWriter
 	TemplateParser   cloudinit.ITemplateParser
 	Recorder         record.EventRecorder
+	Installer        Installer
```

Comment:
```suggestion
	K8Installer        Installer
```
just to be more specific